tests/server/test_elasticsearch.py

```diff
@@ -0,0 +1,117 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import random
+
+import pytest
+from argilla.server.elasticsearch import ElasticSearchEngine
+from argilla.server.models import AnnotationType
+from argilla.server.settings import settings
+from elasticsearch import BadRequestError, Elasticsearch
+from opensearchpy import OpenSearch
+from sqlalchemy.orm import Session
+
+from tests.factories import AnnotationFactory, DatasetFactory
+
+
+def _running_with_elasticsearch() -> str:
+    open_search = OpenSearch(hosts=settings.elasticsearch)
+
+    info = open_search.info(format="json")
+    version_info = info["version"]
+
+    return "distribution" not in version_info
+
+
+@pytest.fixture(scope="session")
+def es_config():
+    return {"hosts": settings.elasticsearch}
+
+
+@pytest.fixture(scope="session")
+def elasticsearch(es_config):
+    client = Elasticsearch(**es_config)
+    yield client
+
+    for index_info in client.cat.indices(index="ar.*,rg.*", format="json"):
+        client.indices.delete(index=index_info["index"])
+
+
+@pytest.fixture(scope="session")
+def search_engine(es_config):
+    return ElasticSearchEngine(config=es_config)
+
+
+@pytest.mark.skipif(
+    condition=not _running_with_elasticsearch(),
+    reason="Test only running with elasticsearch backend",
+)
+def test_create_index_for_dataset(search_engine: ElasticSearchEngine, elasticsearch: Elasticsearch):
```

Comment:
We should not allow to create datasets without annotations so not sure if this case it's important.

Reply:
This validation should be placed at the dataset creation level right? Otherwise, we should replicate the validation for each engine implementation.

Reply:
I can imagine this as a validation step when a dataset becomes ready.

Reply:
> I can imagine this as a validation step when a dataset becomes ready.

Exactly, when a request to publish the dataset is done if the dataset doesn't has annotation the publish procedure should fail.